app/src/main/java/org/simple/clinic/settings/changelanguage/ChangeLanguageLogic.kt

```diff
@@ -0,0 +1,18 @@
+package org.simple.clinic.settings.changelanguage
+
+import com.spotify.mobius.First
+import com.spotify.mobius.First.first
+
+object ChangeLanguageLogic {
+
+  fun init(model: ChangeLanguageModel): First<ChangeLanguageModel, ChangeLanguageEffect> {
+    val effects = when {
+      model.currentLanguage == null && model.supportedLanguages.isEmpty() -> setOf(LoadCurrentSelectedLanguageEffect, LoadSupportedLanguagesEffect)
```

Comment:
How can we make these more descriptive? Can we make why a certain effect is being dispatched more evident?

Reply:
Makes sense, I'll take a look and see what I can do.

Reply:
Extracted these conditions into more descriptive variables. I feel these provide more information now than earlier. Please take a look?